### ISSUE TITLE
nn.DataParallel always creates a "sub-batch" of size 1.

### ISSUE THREAD

**Issue by sanjeevmk** [pytorch/pytorch#18563]
I understand that nn.Dataparallel divides the batch across available GPUs. And it probably tries to do an equal split? However, for some reason, this division somehow always creates a "sub-batch" of size 1, which then leads to failure because my model has Batch Normalization layers. I have 4 GPUs on a system, and I tried with a batch size of 16. The expectation is that the batch will be divided into 4 "sub-batches" of size 4 each or some division with >1 size for each "sub-batch", but that doesn't happen. I tried changing the batch size to different values, but it always creates a "sub-batch" of size 1. The training proceeds as normal, but fails at the very last item of the epoch, because that batch has a size of 1, and BN over features can't be done.

I'm doing this with Python3.5 and Torch '1.0.1.post2'
